app/javascript/packs/order_screen.js

```diff
@@ -63,7 +63,10 @@ document.addEventListener('turbolinks:load', () => {
         },
 
         setUser(user = null) {
-          this.orderRows = [];
+          if (this.selectedUser === null || user === null || this.selectedUser.id != user.id) {
```

Comment:
Makes sure the selected products are remembered if some are selected, and the user's balance is then increased using cash